src/edu/ucsb/nceas/metacat/restservice/v2/MNResourceHandler.java

```diff
@@ -1845,5 +1858,47 @@ protected void updateSystemMetadata() throws ServiceFailure, InvalidRequest,
         MNodeService.getInstance(request).updateSystemMetadata(session, pid, systemMetadata);
     }
 
+    /**
+     * Handle the reindex request
+     * @throws InvalidRequest
+     * @throws ServiceFailure
+     * @throws NotAuthorized
+     * @throws NotImplemented
+     */
+    protected void reindex() throws InvalidRequest, ServiceFailure, NotAuthorized, NotImplemented {
+        boolean all = false;
+        List<Identifier> identifiers = new ArrayList<Identifier>();
+        String[] allValueArray = params.get("all");
+        logMetacat.debug("MNResourceHandler.reindex - the allValueArray in the request is "
+                                                                               + allValueArray);
+        if (allValueArray != null) {
+            if (allValueArray.length != 1) {
+                throw new InvalidRequest("5903", "The \"all\" should only have one vaule");
+            } else {
+                String allValue = allValueArray[0];
+                if (allValue != null && allValue.equalsIgnoreCase("true")) {
+                    all = true;
+                }
+            }
+        }
+        if (!all) {
+            String[] ids = params.get("pid");
+            logMetacat.debug("MNResourceHandler.reindex - the pid list in the request is " + ids);
+            if (ids != null) {
+                for (String id : ids) {
+                    if (id != null && !id.trim().equals("")) {
+                        Identifier identifier = new Identifier();
+                        identifier.setValue(id);
+                        identifiers.add(identifier);
+                    }
+                }
+            } else {
+                throw new InvalidRequest("5903", "Users should specify the \"pid\" vaule "
+                                                                            + "for reindexing");
+            }
+        }
+        MNodeService.getInstance(request).reindex(session, identifiers, all);
```

Comment:
It's almost always bad practice to have a boolean parameter in a public method if that method is not a setter. 

When reading this calling code, it's not obvious what the boolean stands for without looking at the source or documentation. (also known as [the boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap)).

I think it would be much clearer if this code called the `handleReindexAllAction()` or the `handleReindexAction()` directly, instead of passing the boolean.